docs/proposals/image-promotion.md

```diff
@@ -0,0 +1,237 @@
+  * By moving the ImageStream from one project to another
+    - `oc export is -l name=application | oc create -n stage -f -`
+  * By using Docker CLI manually
+    - `docker tag application/image qa-registry:application/image:qa-ready`
+    - `docker push qa-registry:application/image:qa-ready`
+* **Automatic promotion**
+  * By using CLI
+    - Create cron job that tags the image when the image with watched tag is
+      available in registry
+  * By using OpenShift internal mechanism
+    - Use DeploymentConfig lifecycle hooks to do `oc tag`
+    - Use DeploymentConfig lifecycle hooks to do `docker tag` and `docker push`
+    - Use DeploymentConfig to call remote OpenShift API to do tagging
+    - Use DeploymentConfig strategy that deploys a 'testing' image and calls
+      the OpenShift API
+  * By using external Continuous delivery tools
+    - Use Jenkins job to do `oc tag` or call the OpenShift API
+    - Use Jenkins job to do `docker tag` and `docker push`
+* **Semi-automated (require human intervention for some steps)**
+  * By using OpenShift or external Continuos delivery tool
+    - Use Jenkins job to do produce and verify the new Docker image, but require human
+      intervention to import the image to OpenShift
+    - Use Jenkins job to do `oc tag` but have DeploymentConfig independent of
+      the image change (require manual intervention to roll out a new deployment)
+    - Use DeploymentConfig lifecycle hook to do `oc tag` but require human to start a new deployment
+
+## Use Cases
+
+### Automatic promotion
+
+As a developer of Rails application, when I update the source code and push it to a remote
+source code repository, I want to have my changes tested before deploying them to production.
+I also want to have the Docker image that contains applications with my changes to be promoted
+based on the verification results.
+
+#### Example of using Jenkins
+
+As a user I have a Jenkins server running in my project by following the instructions from the [Jenkins example](https://github.com/openshift/origin/blob/master/examples/jenkins/README.md).
+Then I create the *frontend-production* DeploymentConfig in my *Prod* project.
+
+This DeploymentConfig has the ImageChangeTrigger set to watch changes in the "prod-ready" ImageStreamTag:
+
+```json
+{
+  "type": "ImageChange",
+  "imageChangeParams": {
+    "automatic": true,
+    "from": {
+      "kind": "ImageStreamTag",
+      "name": "origin-ruby-example:prod-ready"
+    },
+    "containerNames": [
+      "frontend"
+    ]
+  }
+}
+```
+
+I also have a BuildConfig defined for my application image that does not specify any triggers.
+
+When a change is detected in remote source code repository, the Jenkins job will connect to OpenShift and
+start a new build.
+Based on the build status, Jenkins job then tag the updated application image as *prod-ready*.
+When a new image with this tag is available, the *frontend-production* manages to roll out a new deployment.
```

Comment:
@jorgemoralespou rollback should be possible through deploymentConfig strategy IMHO